### PR TITLE
Support overwriting existing save states

### PR DESCRIPTION
**Pull request by yoshisuga**
Show an alert controller when selecting a save state

### CODE REVIEW

File: Provenance/Emulator/PVEmulatorViewController.swift [hunk 1]
```diff
@@ -721,45 +721,46 @@ class PVEmulatorViewController: PVEmulatorViewControllerRootClass, PVAudioDelega
...
 		let image = captureScreenshot()
 		createNewSaveState(auto: true, screenshot: image)
-	}
-	
+    }
+
 	func createNewSaveState(auto: Bool, screenshot: UIImage?) {
 		let saveFile = PVFile(withURL: URL(fileURLWithPath: saveStatePath).appendingPathComponent("\(game.md5Hash)|\(Date().timeIntervalSinceReferenceDate).svs"))
-		
-		var imageFile: PVImageFile?
```
[the remark this answers is not in the document]
Comment (yoshisuga) [reply Provenance/Emulator/PVEmulatorViewController.swift]: yeah it looks like it - i didn't change any of the logic in create the image file and save state. i must have moved things around and put it back in place and did a ctrl-i on that entire block.

[the remark this answers is not in the document]
Comment (jasarien) [reply Provenance/Emulator/PVEmulatorViewController.swift]: In future, we’d prefer not to have changes like that committed. It makes reviewing the PR more difficult.

[the remark this answers is not in the document]
Comment (yoshisuga) [reply Provenance/Emulator/PVEmulatorViewController.swift]: understood - let me try to revert the changes for that section

[the remark this answers is not in the document]
Comment (jasarien) [reply Provenance/Emulator/PVEmulatorViewController.swift]: That’s much better, thanks!

**Comment by yoshisuga** [comment]
Ok, updated my branch and reverted that unchanged section in the `createNewSaveState` method

Looks like this happened because i was using soft-tabs (spaces instead of tabs) :(

**Comment by jasarien** [comment]
Just gave this a test. Seems to work fine — couldn’t see any issues.

**Pull request merged**